Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt

```diff
@@ -108,6 +111,9 @@ class HomeFragment : Fragment(R.layout.home_fragment_layout), AutoInject {
                 HomeFragmentEvents.ShowReactivateRiskCheckDialog -> {
                     showReactivateRiskCheckDialog()
                 }
+                HomeFragmentEvents.GoToContactJournalDay -> {
+                    (activity as MainActivity).goToContactJournal()
```

Comment:
Why does this logic have to go  through the home fragment, can't the activity directly navigate to the diary?

Reply:
`HomeFragment` already handles `HomeFragmentEvents` such as delta onboarding or new release info and those events have priority before opening the contact journal. Maybe, I can even refactor this behavior and extract those events to `OnboardingActivity` and then I will be able to call the contact journal from `MainActivity`. Or let me know if you have another idea.

Reply:
This point also I think you can avoid using `navigate(uri())`

Reply:
I've experimented with one concept and get cleaner results by refactoring onboarding, let me know your thoughts. It's in PR: https://github.com/corona-warn-app/cwa-app-android/pull/2398
Basically, I'm doing a check for delta onboarding and new features screen before starting `MainActivity` so it's guaranteed that those fragments will be shown before anything else

Reply:
PR has been merged. So this comment is resolved now.